streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -384,14 +381,16 @@ private void closeTopology() {
         // close the processors
         // make sure close() is called for each node even when there is a RuntimeException
         RuntimeException exception = null;
-        for (final ProcessorNode node : topology.processors()) {
-            processorContext.setCurrentNode(node);
-            try {
-                node.close();
-            } catch (final RuntimeException e) {
-                exception = e;
-            } finally {
-                processorContext.setCurrentNode(null);
+        if (taskInitialized) {
```

Comment:
This is for line 379 above: in `suspend` we call `partitionGroup.clear`, and then in closeSuspended we call `partitionGroup.close` and they are the only callers of these groups. So I feel we can remove `queuesByTime.clear()` in `PartitionGroup.close()` as we determine `close()` is always following a `clear()`.